semgrep-core/src/analyzing/AST_to_IL.ml

```diff
@@ -352,16 +372,37 @@ and assign env lhs _tok rhs_exp eorig =
 (* less: we could pass in an optional lval that we know the caller want
  * to assign into, which would avoid creating useless fresh_var intermediates.
  *)
-and expr_aux env eorig =
+and expr_aux env ?(void = false) eorig =
   match eorig.G.e with
-  | G.Call ({ e = G.IdSpecial (G.Op op, tok); _ }, args) ->
+  | G.Call ({ e = G.IdSpecial (G.Op op, tok); _ }, args) -> (
       let args = arguments env args in
-      mk_e (Operator ((op, tok), args)) eorig
+      if not void then mk_e (Operator ((op, tok), args)) eorig
+      else
+        (* The operation's result is not being used, so it may have side-effects.
+         * We then assume this is just syntax sugar for a method call. E.g. in
+         * Ruby `s << "hello"` is syntax sugar for `s.<<("hello")` and it mutates
+         * the string `s` appending "hello" to it. *)
+        match args with
+        | [] -> impossible (G.E eorig)
+        | obj :: args' ->
+            let obj_var, _obj_lval = var_of_exp env tok obj in
+            let method_id = (Parse_info.str_of_info tok, tok) in
+            let method_name = (method_id, -1) in
```

Comment:
Very unimportant, but this would be nice as a record. `-1` doesn't mean anything to me

Reply:
I agree. As one can infer from some of my PR comments, I am not fond of tuple types myself, I tend to prefer record types. Maybe a good small task for the next anti-hack week? That second field is the `sid`, `-1` just says that the name has not been resolved.